### PR TITLE
Create volunteers show endpoint

### PR DESCRIPTION
**Pull request by kylefarmer85**
[ticket](https://trello.com/c/RJbvvxdU/168-voma-create-endpoint-to-retrieve-data-from-volunteer-table)

I created a VolunteersController with a show action to retrieve a single volunteer's data. Also created the corresponding route. 

I created the volunteers controller with an api namespace, so the endpoint is /api/volunteers/:id

### CODE REVIEW

File: app/controllers/api/volunteers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Api::VolunteersController < ApplicationController
+
+  def show
```
Comment (jrapala) [inline app/controllers/api/volunteers_controller.rb]: Can we add a comment that this is for the GET /api/volunteers/:id endpoint?

**Review by jrapala — CHANGES_REQUESTED** [review]

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  namespace :api do
```
Comment (jrapala) [inline config/routes.rb:3]: Can we version this? `v1`?

File: app/controllers/api/v1/volunteers_controller.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Api::V1::VolunteersController < ApplicationController
+
+  # GET /api/vi/volunteers/:id
```
Comment (jrapala) [inline app/controllers/api/v1/volunteers_controller.rb]: Should be `/v1/`

File: app/controllers/api/v1/volunteers_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  def show
+    volunteer = Volunteer.find(params[:id])
+    if volunteer
+      render json: volunteer
+    else
+      render json: {error: "Volunteer not found"}
```
Comment (jrapala) [inline app/controllers/api/v1/volunteers_controller.rb:10]: I did not get this error message anywhere

```
{
    "status": 404,
    "error": "Not Found",
    "exception": "#<ActiveRecord::RecordNotFound: Couldn't find Volunteer with 'id'=0>",
    "traces": {
        "Application Trace": [
            {
                "exception_object_id": 13740,
                "id": 1,
                "trace": "app/controllers/api/v1/volunteers_controller.rb:5:in `show'"
            }
        ],
        "Framework Trace": [
            {
                "exception_object_id": 13740,
                "id": 0,
                "trace": "activerecord (6.1.4) lib/active_record/core.rb:338:in `find'"
            },
            {
```

**Review by jrapala — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/volunteers_controller.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class Api::V1::VolunteersController < ApplicationController
+
+  # GET /api/vi/volunteers/:id
+  def show
+    volunteer = Volunteer.find(params[:id])
+    if volunteer
+      render json: volunteer
```
Comment (jrapala) [inline app/controllers/api/v1/volunteers_controller.rb:8]: Can we camelCase the keys please? 🙏

```
{
    "id": 1,
    "name": "test name",
    "email": "test email",
    "slack_user_id": null,
    "pronouns": null,
    "primary_skills": null,
    "secondary_skills": null,
    "employer": null,
    "student": null,
    "assigned_projects": null,
    "job_title": null,
    "created_at": "2021-07-16T01:06:27.770Z",
    "updated_at": "2021-07-16T01:06:27.770Z"
}
```

**Review by jrapala — DISMISSED** [review]

**Review by Ryan-Koch — APPROVED** [review]

**Pull request merged**